core-bundle/src/Filesystem/FilesystemItem.php

```diff
@@ -44,20 +44,32 @@ class FilesystemItem
      */
     private $extraMetadata;
 
+    /**
+     * @var string|(\Closure(self):string|null)|null
+     */
+    private $publicUri;
+
     /**
      * @param int|(\Closure(self):int|null)|null $lastModified
      * @param int|\Closure(self):int|null $fileSize
      * @param string|\Closure(self):string|null $mimeType
      * @param array<string, mixed>|\Closure(self):array<string, mixed> $extraMetadata
+     * @param string|(\Closure(self):string|null)|null $publicUri
      */
-    public function __construct(bool $isFile, string $path, $lastModified = null, $fileSize = null, $mimeType = null, $extraMetadata = [])
+    public function __construct(bool $isFile, string $path, $lastModified = null, $fileSize = null, $mimeType = null, $extraMetadata = [], $publicUri = null)
     {
         $this->isFile = $isFile;
         $this->path = $path;
         $this->lastModified = $lastModified;
         $this->fileSize = $fileSize;
         $this->mimeType = $mimeType;
         $this->extraMetadata = $extraMetadata;
+
+        // todo: We should probably implement this as a callback instead of a
+        //       parameter, because it will never be known in advance; would
+        //       also allow to pass the options. If options are serializable,
+        //       results could still be cached here.
+        $this->publicUri = $publicUri;
```

Comment:
I don't think this is correct here. The public url should not be in this value object. Just call `$this->vfs->getPublicUri($filesystemItem)`. The `Route` also doesn't know about it's full URI. You call the router for that :)

Reply:
I would call the method(s) `generatePublicUri()` everywhere and then it would effectively just be a callback. The only reason for it being in the `FilesystemItem` would be, because the item is something that can nicely be passed around. But we can still implement this later on if needed, I suppose. :+1:

Reply:
I removed the usage in c9bd99b14c7f66c574a95c7a6d89ca917e2ef0d8